fuse/readonly/readonly_unix.go

```diff
@@ -181,8 +181,11 @@ func (s *Node) Read(req *fuse.ReadRequest, resp *fuse.ReadResponse, intr fs.Intr
 	if err != nil {
 		return err
 	}
-	n, err := io.ReadFull(r, resp.Data[:req.Size])
+	n, err := r.Read(resp.Data[:req.Size])
```

Comment:
Okay, just to recap us to the same page.

`ReadFull()` returns `ErrUnexpectedEOF` if less then `len(resp.Data[:req.Size])` is read from `r`. So that won't work here and give the fuse layer a wrong impression.

For the case with just 5 bytes to read, a single `Read()` call works but what happens when you have 2kb of data in the DAG and the runtime feels like just returning 1kb from `Read()`? You would have to do the bookkeeping that `io.Copy()` is made for. The `io.LimitReader()` is there to not read more from the DAG than the request wants.

I purposed this approach because `io.Copy()` and `io.LimitReader` play well together. `io.Copy()` expects `EOF` from `src` and will return `err==nil` in that case. `io.LimitReader` will also just return `EOF` if it's limit is reached. So only exceptional cases like `broken pipe` will trigger an actual error here.